### PR TITLE
feat: add working_dir option for run action

### LINKED ISSUE #8: Add working directory option for run action
## Summary

When executing commands via the `run` action, the command is executed in the directory where `.claude/hooks-rules.toml` is located. This causes issues when the project structure has `package.json` or other build configuration files in a subdirectory.

## Problem

Given the following directory structure:

```
/workspaces/project/
├── .claude/
│   ├── settings.local.json
│   └── hooks-rules.toml
└── app/
    └── package.json  # contains "format" script
```

With this rule:

```toml
[rules.auto-format-biome]
event = "PostToolUse"
matcher = "Edit|Write"
action = "run"
command = "bun run format"
on_error = "ignore"
when.file_path = "\\.(js|jsx|ts|tsx|json)$"
```

The command `bun run format` fails because it runs in `/workspaces/project/` where no `package.json` exists, instead of `/workspaces/project/app/`.

## Proposed Solution

Add a `working_dir` (or `cwd`) option to specify the directory where the command should be executed:

```toml
[rules.auto-format-biome]
event = "PostToolUse"
matcher = "Edit|Write"
action = "run"
command = "bun run format"
working_dir = "app"  # relative to .claude/ parent directory
# or
# working_dir = "/workspaces/project/app"  # absolute path
on_error = "ignore"
when.file_path = "\\.(js|jsx|ts|tsx|json)$"
```

## Alternatives Considered

- Using `cd app && bun run format` in the command - works but is less clean
- Using absolute paths like `bunx biome format --write` - works but loses package.json script benefits

### PR DESCRIPTION
**Pull request by 1gy**
## Summary

Add support for specifying the working directory when executing commands via the `run` action.

- Add `working_dir` option to specify command execution directory
- Add `${file_dir}` variable (parent directory of `file_path`)
- Add `${workspace_root}` variable (cchooked's CWD)
- Default working directory is now `file_path`'s parent directory
- Relative paths are resolved from `workspace_root`

## Usage

```toml
# Default: runs in file_path's parent directory
[rules.auto-format]
command = "bun run format"

# Explicit workspace root
[rules.build]
command = "bun run build"
working_dir = "${workspace_root}"

# Subdirectory
[rules.frontend-lint]
command = "bun run lint"
working_dir = "${workspace_root}/frontend"
```

## Test plan

- [x] Unit tests for new template variables
- [x] Integration tests for working_dir option
- [x] Edge case tests (no file_path, nonexistent directory, empty expansion)
- [x] Manual verification with Claude Code hooks

Closes #8

### CODE REVIEW

File: src/action.rs [hunk 1]
```diff
@@ -6,6 +6,28 @@ use std::fs::OpenOptions;
...
+                std::path::PathBuf::from(&context.workspace_root)
+                    .join(&expanded)
+                    .to_string_lossy()
+                    .into_owned()
+            }
+        }
+        None => context.file_dir.clone(),
+    };
+
+    if dir.is_empty() { None } else { Some(dir) }
```
[the remark this answers is not in the document]
Comment (1gy) [reply src/action.rs]: Fixed. Refactored with explicit nested if-else to clarify the double-fallback logic.

File: src/action.rs [hunk 2]
```diff
@@ -6,6 +6,28 @@ use std::fs::OpenOptions;
...
+    let dir = match working_dir {
+        Some(template) => {
+            let expanded = context.expand(template);
+            if expanded.is_empty() {
+                context.file_dir.clone()
+            } else if std::path::Path::new(&expanded).is_absolute() {
+                expanded
+            } else {
+                std::path::PathBuf::from(&context.workspace_root)
+                    .join(&expanded)
```
[the remark this answers is not in the document]
Comment (1gy) [reply src/action.rs]: Fixed. Added fallback to `file_dir` when `workspace_root` is empty.
